ledger/src/blockstore_processor.rs

```diff
@@ -328,9 +328,6 @@ fn execute_batches(
 
     let mut minimal_tx_cost = u64::MAX;
     let mut total_cost: u64 = 0;
-    // Allowing collect here, since it also computes the minimal tx cost, and aggregate cost.
-    // These two values are later used for checking if the tx_costs vector needs to be iterated over.
-    #[allow(clippy::needless_collect)]
```

Comment:
nit - maybe keep the comment? When I first looked at this code, i thought why collect here, iterator should be enough; then the comment pointed out it also computes local variables. So in that sense, it's useful; (or refactor Ln 339-340 our)

Reply:
Fair, my thinking was that with your changes in place, we do explicitly use the `tx_costs` later on. So even tho we are updating variables as we iterate through, we are going to explicitly use the collected vector as well

Reply:
Gotcha, think my code below can use mapped iterators, doesn't really need vec<Tx_cost> being collected first. But I can see now that comments can be redundant here. No problem to remove it. Can do refactor in other time.

Reply:
`CostModel::calculate_cost()` returns a `TransactionCost` that has non-trivial members (namely that `Vec<Pubkey>`).
https://github.com/solana-labs/solana/blob/2ba73270270d037ef1d7dba71a8476bce77868a4/runtime/src/cost_model.rs#L31-L40

Once your newly added feature becomes active, we might iterate through `tx_costs` (the Vec that results from the `collect()`) twice. If we remove the `collect()`, we would have to calculate / allocate each individual `TransactionCost` each time we iterate through with `map()` right? The implementation of `CostModel::calculate_cost()` look non-trivial enough that it seems avoiding this might be preferable.